README.md

```diff
@@ -809,6 +864,173 @@ query = EnvironmentQuery(name="name",
 ```
 
 
+### Collectors
+
+The post-build step has access to the build log containing the list of modules
+that were built and what files they generated.
+However, these modules also need to pass additional data to the post-build steps,
+so that this information can be computed locally.
+
+*lbuild* allows each module to declare what metadata it wants using a collector,
+which is given a name, description and optional limitations depending on type.
+In the build step, each module may add values to this collector, which the
+post-build steps then can access.
+
+```python
+def prepare(module, options):
+    # Add a collector to module
+    collector = Collector(...)
+    module.add_collector(collector)
+
+def build(env):
+    exists = env.has_collector(":module:collector")
+    # Add values to this collector
+    env.collect(":module:collector", "value1", "value2", "value3")
+
+def post_build(env):
+    # Get all unique values from all scopes
+    unique_values = env.collector_values(":module:collector")
+    # get all values from all scopes, even duplicates!
+    all_values = env.collector_values(":module:collector", unique=False)
+```
+
+Note that the ordering of values is preserved only relative to the order they
+were added to the same scope within a module!
+The above example will preserve the order of `value1`, `value2` and `value3`,
+only if the values are accessed not uniquely and only relative to each other.
+
+When you add values to a collector, the current scope is recorded, consisting
```

Comment:
The name `scope` is not wrong, but too much coupled to the lexical scope of the Python program. At least for me. Maybe `context` is better.

Reply:
Hm, but context is so generic…

Reply:
The same is true for scope :smile: 

For me *scope* was confusing without looking into the implementation. I would associate other things with it.

Reply:
Yes, I agree. Maybe `operations` covers it more closely in the scope of this context (ha!), and technically it is quite close to the buildlog operations. I've actually intentionally made the `BuildLogOperationFacade` have the same functions as the `Scope` class, so maybe we should just use the semantics of the OperationFacade for this?

Reply:
Implemented it this way now, and it's better.